tutorials/2d/custom_drawing_in_2d.rst

```diff
@@ -96,7 +97,145 @@ call update() from the _process() callback, like this:
+
+::
+
+    func _draw():
+        var center = Vector2(200,200)
+        var radius = 80
+        var angleFrom = 75
+        var angleTo = 195
+        var color = Color(1.0, 1.0, 1.0)
+        draw_circle_arc( center, radius, angleFrom, angleTo, color )
+
+Result:
+
+.. image:: /img/result_drawarc.png
+
+
+Arc polygon function
+^^^^^^^^^^^^^^^^^^^^
+
+Only for example purpose, we also provide here another function, very similar, that draws the plain portion of the disc defined by the arc, not only its shape. The method is exactly the same a previously, except that we draw a polygon instead of lines.
+
+::
+
+    func draw_circle_arc_poly( center, radius, angleFrom, angleTo, color ):
+        var nbPoints = 32
+        var pointsArc = Vector2Array()
+        pointsArc.push_back(center)
+        var colors = ColorArray([color])
+    
+        for i in range(nbPoints+1):
+            var anglePoint = angleFrom + i*(angleTo-angleFrom)/nbPoints - 90
+            pointsArc.push_back(center + Vector2( cos( deg2rad(anglePoint) ), sin( deg2rad(anglePoint) ) ) * radius)
+        draw_polygon(pointsArc, colors)
+        pass
+
+Dynamic custom drawing
+^^^^^^^^^^^^^^^^^^^^^^
+Alright, we are now able to draw custom stuff on screen. However, it is very static: let's make this shape turn around the center. The solution to do this is simply to change the angleFrom and angleTo values over time. For our example, we will simply increment them by 50. This increment value has to remain constant, else the rotation speed will change accordingly.
+
+First, we have to make both angleFrom and angleTo variables global at the top of our script. Also note that you can store them in other nodes and access them using get_node().
+
+::
+
+ extends Node2D
+
+ var rotationAng = 50
+ var angleFrom = 75
+ var angleTo = 195
+
+
+
+We make these values change in the _process(delta) function. To activate this function, we need to call set_process(true) in the _ready() function. 
+
+We also increment our angleFrom and angleTo values here. However, we must not forget to clamp() the resulting values between 0 and 360°! If you don't, the script will work correctly but angles values will grow bigger and bigger over time, until they reach the maximum integer value Godot can manage (2^31 - 1). When this happens, Godot may crash.
+
+Finally, we must not forget to call the update() function, which automatically calls _draw(). This way, you can control when you want to refresh the frame.
+
+::
+
+ func _ready():
+     set_process(true)
+
+ func _process(delta):
+     angleFrom += rotationAng
+     angleTo += rotationAng
+     angleFrom = clamp(angleFrom, 0, 360)
+     angleTo = clamp(angleTo, 0, 360)
+     update()
+
+Also, don't forget to modify the _draw() function to make use of these variables:
+::
+
+ func _draw():
+	var center = Vector2(200,200)
+	var radius = 80
+	var color = Color(1.0, 0.0, 0.0)
+
+	draw_circle_arc( center, radius, angleFrom, angleTo, color )
+
+Let's run!
```

Comment:
Note that I don't think that reST would display line 222 and 223 as two lines, it will likely remove the newline if there's no empty newline to mark two paragraphs. You can use the following formatting to achieve a single block of various lines, though we'd better not use it too much IMO as it produces a peculiar formatting IIRC:
```
| Let's run!
| It works, but the arc is rotating insanely fast! What's wrong?
```